odoo/orm/fields_relational.py

```diff
@@ -1393,18 +1393,23 @@ def read(self, records):
 
         # retrieve pairs (record, line) and group by record
         group = defaultdict(list)
+        corecord_ids = OrderedSet()
         for id1, id2 in records.env.execute_query(query.select(sql_id1, sql_id2)):
             group[id1].append(id2)
+            corecord_ids.add(id2)
 
         # filter using record rules
-        if filter_access and group:
-            corecord_ids = OrderedSet(id_ for ids in group.values() for id_ in ids)
+        if filter_access and corecord_ids:
```

Comment:
nitpicking: Can you rename the variable `filter_access` name by `bypass_access` ?